### PR TITLE
Fixed deprecation on YAML component

### PR DESCRIPTION
**Pull request by Shine-neko**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Fixed YAML deprecation and refactor Twig global (headerTabContent) |
| Type? | bug fix |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |

### CODE REVIEW

**Comment by maximebiloe** [comment]
Can you make sure to respect the commit standards? :smile:

**Comment by Shine-neko** [comment]
@maximebiloe  Done

**Comment by mickaelandrieu** [comment]
@Shine-neko still not good :) 

~~``Co: Fix dependencies``~~ => ``CO: Fix dependencies``

File: src/PrestaShopBundle/Resources/config/admin/services.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ services:
             - "@prestashop.hook.dispatcher"
     prestashop.adapter.admin.wrapper.product:
         class: PrestaShop\PrestaShop\Adapter\Product\AdminProductWrapper
-        arguments: ["@translator", "@prestashop.adapter.legacy.context"]
+        arguments: ["@prestashop.adapter.translator", "@prestashop.adapter.legacy.context"]
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/config/admin/services.yml]: should be ``@translator`` atm

File: src/PrestaShopBundle/Twig/AdminExtension.php [hunk 2]
```diff
@@ -23,41 +24,53 @@
...
+namespace PrestaShopBundle\Twig;
+
 use Symfony\Component\Yaml\Parser;
-use Symfony\Component\Routing\Router;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\ParameterBag;
 
-class AdminControllersListener
+/**
+ * Twig extension for the Symfony Asset component.
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Twig/AdminExtension.php]: not realy

We should read instead ``Twig extension to share globaly the admin header menu``.

File: src/PrestaShopBundle/Twig/AdminExtension.php [hunk 3]
```diff
@@ -66,23 +79,44 @@ public function onKernelRequest(GetResponseEvent $event)
...
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getGlobals()
+    {
+        $globals = array();
+
+        if (null !== $this->requestStack->getCurrentRequest()) {
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Twig/AdminExtension.php]: hmm you can do even better: only build and return it in "admin" back office.
Sounds like a premature optimization but I'm pretty sure no one will come back here ^_^

But sounds like boring as you need to load ``admin_routing.yml`` file and then try to match the current route with routerMatcher: I'm not sure it's worth from a performance point of view: what do you think ?

Comment (mickaelandrieu) [inline src/PrestaShopBundle/Twig/AdminExtension.php]: Better idea, make ``$tabDataContent`` static and reuse it: this menu won't change during the same request.

Comment (Shine-neko) [inline src/PrestaShopBundle/Twig/AdminExtension.php]: I gree. I just migrated to remove the deprecated stuff

**Pull request merged**